Tests/DeviceDetectorTest.php

```diff
@@ -460,6 +460,63 @@ public function testCheckRegexRestrictionEndCondition(): void
         $this->assertTrue($this->checkRegexRestrictionEndCondition('TestValue(?:[;)/]|$)'), 'pass condition');
     }
 
+    public function testCheckYamlExtraLint(): void
+    {
+        $files = \array_merge(
+            \glob(__DIR__ . '/../regexes/client/*.yml'),
+            \glob(__DIR__ . '/../regexes/device/*.yml')
+        );
+
+        foreach ($files as $file) {
+            $fn = \fopen($file, 'r');
+            $n  = 0;
+
+            while (!\feof($fn)) {
+                $line = \fgets($fn);
+
+                if (false === $line) {
+                    break;
+                }
+
+                if (\preg_match('~^\s+\d+\.(?:[\d\.]+)\:~si', $line)) {
+                    $message = \sprintf(
+                        "Key is number format, need wrap key in string. Example \"'number': 'value'\"\n" .
+                        "File: %s\nLine: %s\nString: %s",
+                        $file,
+                        $n,
+                        $line
+                    );
+                    $this->assertTrue(false, $message);
+                }
+
+                $n++;
+            }
+
+            \fclose($fn);
+        }
+
+        $this->assertTrue(true);
+    }
+
+    /**
+     * check the Symfony parser for fixtures parsing errors
+     */
+    public function testSymfonyParser(): void
+    {
+        $files       = \array_merge(
+            \glob(__DIR__ . '/../regexes/client/*.yml'),
+            \glob(__DIR__ . '/../regexes/client/*.yml'),
+            \glob(__DIR__ . '/../regexes/*.yml')
+        );
+        $yamlSymfony = new Symfony();
+
+        foreach ($files as $file) {
+            $yamlSymfony->parseFile($file);
+        }
+
+        $this->markTestIncomplete();
+    }
```

Comment:
@sgiehl the error occurs only on keys with fractions. on keys `5.2.0`, the parser does not crash.

which of these tests should we leave?

Reply:
Guess would be enough to keep the one that directly uses the YAML parser from Symfony.